commands/provider/marketplace_provider.go

```diff
@@ -0,0 +1,175 @@
+}
+
+// TransferServiceOwnership executes the task transfer service ownership.
+func (p *MarketplaceProvider) TransferServiceOwnership(sid, newOwner, from string) (Transaction, error) {
+	input := marketplaceTransferServiceOwnershipTaskInputs{
+		marketplaceTransactionTaskInputs: marketplaceTransactionTaskInputs{From: from},
+		Sid:                              sid,
+		NewOwner:                         newOwner,
+	}
+	var output Transaction
+	return output, p.call("transferServiceOwnership", input, &output)
+}
+
+// SendSignedTransaction executes the task send signed transaction.
+func (p *MarketplaceProvider) SendSignedTransaction(signedTransaction string) (TransactionReceipt, error) {
+	input := marketplaceSendSignedTransactionTaskInputs{
+		SignedTransaction: signedTransaction,
+	}
+	var output TransactionReceipt
+	return output, p.call("sendSignedTransaction", input, &output)
+}
+
+// GetService executes the task get service.
+func (p *MarketplaceProvider) GetService(sid string) (MarketplaceService, error) {
+	input := marketplaceGetServiceTaskInputs{
+		Sid: sid,
+	}
+	var output MarketplaceService
+	return output, p.call("getService", input, &output)
+}
+
+// IsAuthorized executes the task IsAuthorized.
+func (p *MarketplaceProvider) IsAuthorized(sid string, versionHash string, addresses []string) (bool, string, string, string, error) {
+	input := marketplaceIsAuthorizedInputs{
+		VersionHash: versionHash,
+		Sid:         sid,
+		Addresses:   addresses,
+	}
+	var output marketplaceIsAuthorizedSuccessOutput
+	return output.Authorized, output.Sid, output.Source, output.Type, p.call("isAuthorized", input, &output)
+}
+
+// UploadServiceFiles upload the tarball and the definition file, and returns the address of the definition file
+func (p *MarketplaceProvider) UploadServiceFiles(path string, manifest MarketplaceManifestData) (protocol string, source string, err error) {
+	// upload service source to IPFS
+	tar, err := archive.TarWithOptions(path, &archive.TarOptions{
+		Compression: archive.Gzip,
+	})
+	if err != nil {
+		return "", "", err
+	}
+	tarballResponse, err := ipfs.Add("tarball", tar)
+	if err != nil {
+		return "", "", err
+	}
+
+	manifest.Service.Deployment.Type = marketplaceDeploymentType
+	manifest.Service.Deployment.Source = tarballResponse.Hash
+
+	// upload manifest
+	manifestData, err := json.Marshal(manifest)
+	if err != nil {
+		return "", "", err
+	}
+	definitionResponse, err := ipfs.Add("manifest", bytes.NewReader(manifestData))
+	if err != nil {
+		return "", "", err
+	}
+
+	return marketplaceDeploymentType, definitionResponse.Hash, nil
+}
+
+func (p *MarketplaceProvider) CreateManifest(path string, hash string) (MarketplaceManifestData, error) {
+	var data MarketplaceManifestData
+	definition, err := importer.From(path)
+	if err != nil {
+		return data, err
+	}
+	data.Version = marketplacePublishVersion
+	data.Service.Hash = hash
+	data.Service.HashVersion = "1" // hardcoded for now
```

Comment:
Can be moved in `marketplace_type` file with other constants like the `marketplacePublishVersion`

Reply:
Done